adapter/src/utils/useLocale.js

```diff
@@ -23,22 +52,41 @@ const setGlobalLocale = (locale) => {
     }
     moment.locale(locale)
 
-    const simplifiedLocale = simplifyLocale(locale)
-    i18n.changeLanguage(simplifiedLocale)
+    const resolvedLocale = validateLocaleByBundle(locale)
+    i18n.changeLanguage(resolvedLocale)
+
+    console.log('🗺 Global d2-i18n locale initialized:', resolvedLocale)
+}
+
+// Sets the global direction based on the app's configured direction
+// (which should be done to affect modals, alerts, and other portal elements),
+// then returns the locale's direction for use on the header bar
+const handleDirection = ({ locale, configDirection }) => {
+    // for i18n.dir, need JS-formatted locale
+    const jsLocale = transformJavaLocale(locale)
+    const localeDirection = i18n.dir(jsLocale)
```

Comment:
Finally could we perhaps split all this part (getGlobalDirection + transformJavaLocale + i18n.dir) into another file .. just keep useLocale simple